zb-db/src/main/java/io/zeebe/db/impl/rocksdb/ZeebeRocksDbFactory.java

```diff
@@ -113,26 +116,132 @@ private ZeebeRocksDbFactory(
       throw new IllegalStateException(
           String.format(
               "Expected to create column family options for RocksDB, "
                   + "but one or many values are undefined in the context of RocksDB "
-                  + "[Compiled ColumnFamilyOptions: %s; User-provided ColumnFamilyOptions: %s]. "
+                  + "[User-provided ColumnFamilyOptions: %s]. "
                   + "See RocksDB's cf_options.h and options_helper.cc for available keys and values.",
-              columnFamilyOptionProps, userProvidedColumnFamilyOptions));
+              userProvidedColumnFamilyOptions));
     }
     return columnFamilyOptions;
   }
+
+  private DBOptions createDefaultDbOptions(final List<AutoCloseable> closeables) {
+    final var dbOptions =
+        new DBOptions()
+            .setCreateMissingColumnFamilies(true)
+            .setErrorIfExists(false)
+            .setCreateIfMissing(true)
+            .setParanoidChecks(true)
+            // disabling mmap helps improve performance/memory usage when a DB is highly
+            // fragmented with many small files
+            .setAllowMmapReads(false)
+            .setAllowMmapWrites(false)
+            .setMaxOpenFiles(rocksDbConfiguration.getMaxOpenFiles())
+            // do not hog processing if the DB is highly fragmented by opening many threads to
+            // open all files
+            .setMaxFileOpeningThreads(1)
+            // 1 flush, 1 compaction
+            .setMaxBackgroundJobs(2)
+            // may not be necessary when WAL is disabled, but nevertheless recommended to avoid
+            // many small SST files
+            .setAvoidFlushDuringRecovery(true)
+            // limit the size of the manifest (logs all operations), otherwise it will grow
+            // unbounded
+            .setMaxManifestFileSize(256 * 1024 * 1024L)
+            // keep 1 hour of logs - completely arbitrary. we should keep what we think would be
+            // a good balance between useful for performance and small for replication
+            .setLogFileTimeToRoll(Duration.ofMinutes(30).toSeconds())
+            .setKeepLogFileNum(2);
+
+    // limit I/O writes
+    if (rocksDbConfiguration.getIoRateBytesPerSecond() > 0) {
+      final RateLimiter rateLimiter =
+          new RateLimiter(rocksDbConfiguration.getIoRateBytesPerSecond());
+      dbOptions.setRateLimiter(rateLimiter);
+    }
+
+    if (rocksDbConfiguration.isStatisticsEnabled()) {
+      final var statistics = new Statistics();
+      closeables.add(statistics);
+      statistics.setStatsLevel(StatsLevel.ALL);
+      dbOptions
+          .setStatistics(statistics)
+          // speeds up opening the DB
+          .setSkipStatsUpdateOnDbOpen(true)
+          // can be disabled when not profiling
+          .setStatsDumpPeriodSec(20);
+    }
+
+    return dbOptions;
+  }
+
+  private TableFormatConfig createTableFormatConfig() {
+
+    return new BlockBasedTableConfig()
```

Comment:
I don't see any issues with adding a bloom filter here - it will help with performance (theoretically) with little downsides, no? Let's say we have a 1% false positive rate, i.e. 10 bits for the filter, storing 1,000,000 items would take up ~1200KB, so I don't see much downsides. It's also unclear if it would help the performance at all, but from my pov it can't hurt

Reply:
I guess we can add bloom filter configs also.